### ISSUE TITLE
use native input types (on mobile)

### ISSUE THREAD

**Issue by Haroenv** [iRail/hyperRail#228]
In smaller screens the custom inputs don't really work, since they're too small. However an `<input type="date">` and `<input type="time>` are well supported and will offer a better experience on small screens.

**Comment by pietercolpaert** [comment]
@xavez what's your opinion? 

@Haroenv Can we have a demo implementation of that in hyperRail to see how that would work?

**Comment by xavez** [comment]
I think this should definitely change, but I don’t know enough about angular to do it myself. Ideally, the following happens:

- Check for support of date input type with modernizr or vanilla JS
- Use native date input type if it is supported
- Load JS and CSS for a date picker if they are not supported

Concerning the time input, I think a custom implementation is probably the most user friendly. I’ve implemented this once:

![](http://c.xvr.io/3q3d222H2l0g/Screen%20Recording%202016-08-30%20at%2009.51%20AM.gif)

**Comment by Haroenv** [comment]
I like how the time input works on iOS though, it's just one input to open vs two. Screenshot for reference:

![](http://i.imgur.com/tnDtVfb.jpg)

**Comment by xavez** [comment]
@Haroenv correct, actually, I was mistaken in my comment, the same principle applies there: check for support, otherwise go with custom implementation

**Comment by Haroenv** [comment]
Progressive enhancement would consider desktop chrome as having support for `input type=date` though, I'm not sure if that's what we want ...

**Comment by xavez** [comment]
It’s an improvement over the current situation.

@pietercolpaert do you have usage statistics for irail.be? Browser, Browser Version & Hardware?

**Comment by pietercolpaert** [comment]
Fiddle which enables/disables when input type date is supported/unsupported: http://jsfiddle.net/2AaFk/1/ → I think on a desktop chrome, the input field looks horrible. It looks "okay-ish" on firefox. Is there a way to style this in a nicer way?

## Analytics for browsers and whether or not on a mobile phone

![screenshot from 2016-08-30 12-18-41](https://cloud.githubusercontent.com/assets/347073/18085585/0c3952f6-6eac-11e6-82e4-5f1ae497a800.png)

**Comment by Haroenv** [comment]
I think we should only use the native input method for Android and iOS, since the desktop implementations look pretty bad IMO

**Comment by xavez** [comment]
You can check for touch events and load the appropriate stuff then. 

But like I said, anything is better than the current implementation.

**Comment by xavez** [comment]
Who knows how to make this work in Angular? I can make the "prototype" code, but I have no idea how to bind the input to the search etc.

**Comment by pietercolpaert** [comment]
@xavez it's pretty straightforward. I have made a mock-up in this branch: https://github.com/iRail/hyperRail/tree/fix-228

What I had to change: https://github.com/iRail/hyperRail/compare/development...fix-228

**Comment by xavez** [comment]
Oh cool. Guess I’m not needed for this then ;) 👍

**Comment by pietercolpaert** [comment]
@xavez how do you mean? I implemented this for you to experiment with UX ;) Right now the UX in this branch on both phone and desktop is bad, yet maybe it's possible to make it better?

**Comment by xavez** [comment]
Great, I’ll have a look!
